langchain4j-core/src/main/java/dev/langchain4j/data/document/DocumentParser.java

```diff
@@ -6,7 +6,7 @@
  * Defines the interface for parsing an {@link InputStream} into a {@link Document}.
  * Different document types require specialized parsing logic.
  */
-public interface DocumentParser {
+public interface DocumentParser<T extends Document> {
```

Comment:
Why is this required?

Reply:
To introduce different types of document in the future.

Reply:
Could you please elaborate? What other types do you foresee?